integration-testing/src/tools/rnode.py

```diff
@@ -168,3 +171,24 @@ def create_peer(i, key_pair):
 
     return [ create_peer(i, key_pair)
              for i, key_pair in enumerate(key_pairs)]
+
+
+@contextmanager
+def create_bootstrap(docker, docker_network, timeout, validators_data):
+    bonds_file, bootstrap_keys, _ = validators_data
+    node = create_bootstrap_node(docker, docker_network, bonds_file, bootstrap_keys, timeout)
+
+    yield node
+
+    node.cleanup()
```

Comment:
Again, is this immune to raising exceptions within the `with` body?

Reply:
Yes. A context manager will call the code after the `yield` if an exception is thrown within the `with` statement where it is used